core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/DetailedSampleBuilder.java

```diff
@@ -824,17 +827,14 @@ public void setNonStandardAlias(boolean nonStandardAlias) {
     this.nonStandardAlias = nonStandardAlias;
   }
 
-  public DetailedSample build() {
+  public DetailedSample build(InferStockFromAliquot inferencer) {
```

Comment:
The caller is capable of making it unnecessary. To infer in a nice way requires DAO access, which the caller may not have